### PR TITLE
Added DAG and script to upload sentinel-2 data from nci to s3 bucket

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
https://gajira.atlassian.net/secure/RapidBoard.jspa?rapidView=549&projectKey=DEADM&modal=detail&selectedIssue=DEADM-1270

### CODE REVIEW

File: dags/nci_s2_upload_s3.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+This DAG takes following input parameters:
+
+ * s3bucket: Name of the S3 bucket (e.g. 'dea-public-data-dev')
+ * numdays: Number of days to process before the end date (e.g. '1')
+ * enddate: End date for processing (e.g. '2020-02-21')
+ * doupdate: Check to update granules if already exist. Set 'yes' to update else 'no' or don't set (e.g. 'no')
+
+"""
+
+from datetime import datetime,timedelta
```
Comment (alexgleith) [inline dags/nci_s2_upload_s3.py:19]: There should be a space between these.

Let's run it through a linter...

File: dags/nci_s2_upload_s3.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+from textwrap import dedent
+import os
+
+from airflow import DAG, AirflowException
+from airflow.contrib.hooks.aws_hook import AwsHook
+from airflow.contrib.operators.ssh_operator import SSHOperator
+from airflow.contrib.operators.sftp_operator import SFTPOperator
+from airflow.contrib.operators.sftp_operator import SFTPOperation
```
Comment (alexgleith) [inline dags/nci_s2_upload_s3.py:27]:
```suggestion
from airflow.contrib.operators.sftp_operator import SFTPOperator, SFTPOperation
```

File: dags/nci_s2_upload_s3.py (hunk 1 above)
Comment (alexgleith) [inline dags/nci_s2_upload_s3.py:19]:
```suggestion
from datetime import datetime, timedelta
```

File: dags/nci_s2_upload_s3.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    'nci_s2_upload_s3',
+    doc_md=__doc__,
+    default_args=default_args,
+    catchup=False,
+    schedule_interval=None,
+    template_searchpath='templates/',
+    max_active_runs=1,
+    default_view='graph',
+    tags=['nci', 'sentinel_2'],
+    params={'workdir': '/g/data/v10/work/s2_nbar_rolling_archive/ts_nodash_' + default_args['params']['enddate'] + '_' + default_args['params']['numdays']}
```
Comment (alexgleith) [inline dags/nci_s2_upload_s3.py:56]: I think this should use a string format for clarity and split over multiple lines.

File: dags/nci_s2_upload_s3.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            set -eux
+            {% set work_dir = params.workdir -%}
+            mkdir -p {{work_dir}}
+        '''),
+    )
+    # Uploading s2_to_s3_rolling.py script to NCI
+    sftp_s2_to_s3_rolling_script = SFTPOperator(
+        task_id='sftp_s2_to_s3_rolling_script',
+        local_filepath=os.path.abspath('./scripts/s2_to_s3_rolling.py'),
+        remote_filepath=dag.params['workdir']+'/s2_to_s3_rolling.py',
```
Comment (alexgleith) [inline dags/nci_s2_upload_s3.py:74]:
```suggestion
        remote_filepath=dag.params['workdir'] + '/s2_to_s3_rolling.py',
```

**Comment by alexgleith** [comment]
What is the `requirements.txt` file used for?

**Comment by sachitrajbhandari** [comment]
> What is the `requirements.txt` file used for?

Necessary dependency packages needed in docker to run airflow

**Comment by alexgleith** [comment]
> > What is the `requirements.txt` file used for?
> 
> Necessary dependency packages needed in docker to run airflow

Oh I see, it's mounted as a volume in the Docker Compose. Thanks!

File: dags/nci_s2_upload_s3.py (hunk 1 above)
Comment (sachitrajbhandari) [reply dags/nci_s2_upload_s3.py]: Fixed

File: dags/nci_s2_upload_s3.py (hunk 3 above)
Comment (sachitrajbhandari) [reply dags/nci_s2_upload_s3.py]: Fixed

**Pull request merged**